### ISSUE TITLE
Selection criteria of wallets

### ISSUE THREAD

**Issue by MineForeman** [bitcoin/bitcoin.org#747]
This was originally brought up [elsewhere](https://github.com/bitcoin/bitcoin.org/issues/746#issuecomment-74394704).

Currently only wallets that meet a growing list of [best practice criteria](https://github.com/bitcoin/bitcoin.org#wallets) are accepted for inclusion.  The author of the wallet also has the option to "opt-out" of the list.

I think that the only criteria for inclusion should be that the wallet is used by a significant amount of bitcoin users and there not be an opt-out option.  (How we decide on "significant" is still an issue)

Why I think this is I think we should be informing users of common wallets (warts and all) instead of recommending something based on a set of arbitrarily decided upon best practice criteria (that's how i perceive it).

A case in particular is, what is probably the most used bitcoin wallet is missing from the list because of a lot of known issues. I really think we should be informing users about the not-so-good players with a pile of red marks beside their names.  Not opinions, just the facts.

Anyway, those are my thoughts and it is a total about face of how we do things now.  There are probably a lot of issues with it but I thought I would bring it up anyway to see if anyone else had anything else to say.

**Comment by luke-jr** [comment]
It sounds like what you want is a review website, but I think new users go to bitcoin.org for recommendations on good options more than reviews. Perhaps having both available might make sense - the default could show only wallets meeting best practices *and* having significant use, with options to show problematic or new/unpopular wallets. Although splitting on popular vs unpopular is more likely to lead to centralisation, too... so not so sure about that.

**Comment by MineForeman** [comment]
@luke-jr A review site is the exact opposite of what I mean ;) .

I have probably expressed my ideas badly but what I am angling for is something that includes the bad as well as the good in order to inform the user.

Too many new users are just picking a wallet because everyone else is doing it.  I think bitcoin.org should be informing people about their choice not presenting a list of wallets we think are OK.

**Comment by gmaxwell** [comment]
Might be best to first write something like that separately from Bitcoin.org; and if it works out bitcoin.org can either link to it or include the information.

**Comment by gurnec** [comment]
Starting with the assumption that The Bitcoin Foundation is interested in promoting Bitcoin, I don't see why any page on bitcoin.org shouldn't be limited by an "arbitrary" list of requirements according to its own goals and biases.

In particular, the choose-your-wallet page is targeted at *beginners*. (If it were otherwise, I'd probably agree with you.) As such, it's probably perceived as a list of recommendations by beginners, even if it's not intended to be one. I'm not very optimistic that beginners will take the time to review each wallet for its pros and cons, even if those cons are highlighted on that page. If there's a chance that this page encourages usage of wallets which are more likely to later expose users to theft (and expose Bitcoin to bad press in the process) that seems contrary to The Foundation's goals.

Regarding "arbitrary" requirements: they're simply the best that can be done in an effort to meet these goals. They're surely imperfect, and they are open to challenge. They're also very likely based on the opinions of those who are more Bitcoin-educated than your average Bitcoin beginner.

TLDR: since it's targeted at beginners, I think it's appropriate as-is. If it were an attempt at listing all available wallets, I'd agree with MineForeman, and it'd be a good addition aside the existing page IMO.

**Comment by MineForeman** [comment]
Perhaps I am going about this the wrong way.

I am just thinking that as well as the recommendations, we should have information on wallets that fall at the other end of the spectrum.  It seems that many users will go that way just because they are popular regardless of them being listed here.

**Comment by gurnec** [comment]
@MineForeman Listing wallets on a beginner-oriented choose-your-wallet page which don't follow security best practices (a loose term that's open for interpretation to be sure) doesn't seem responsible. Listing them in an unrelated simple-list-of-all-known-wallets along with a security disclaimer seems somewhat OK to me.

By "other end of the spectrum", were you suggesting listing them somewhere under a heading of dangerous-dont-use, or even simply listing them with lots of red marks in various categories? That seems unwise; it may open up the Foundation to litigation....

**Comment by MineForeman** [comment]
@gurnec Having a list of recommended wallets opens the Foundation to litigation.

Having a list of wallets (good and bad) for the user is the only way to avoid that, it is disseminating information where all candidates have the same criteria applied.

Still, I am not getting my ideas across well, I think what we have now is a review of the best wallets according to us, I think it needs to be more balanced (somehow).  I will keep thinking!

**Comment by luke-jr** [comment]
@gurnec The Bitcoin Foundation only funds bitcoin.org. The site is not owned or operated by the Foundation, and is not intended to serve Foundation goals (whic AIUI are now only focussed on development anyway).

@MineForeman So you want a "list of wallets you should probably not use"? This seems like it would be controversial, so I would prefer keeping it off bitcoin.org; maybe a page on the wiki listing out problems with each?

If litigation is possible, I would think it is for a negative list, certainly not for a recommendation list. Nobody is obliged to recommend one party over or equal to another.

**Comment by MineForeman** [comment]
> If litigation is possible, I would think it is for a negative list, certainly not for a recommendation list. Nobody is obliged to recommend one party over or equal to another.

I was thinking of the "You recommended this wallet and they ran off with all my funds so I am going to sue you".  It is unfortunate, but it happens.  Whereas if we tag a wallet with "Not Audited" and "Non-HSTS" they are just facts.

Anyway, judging from reaction I am barking up the wrong tree, I will think more about it.

**Comment by luke-jr** [comment]
@MineForeman Ah, yes, I guess there is some risk there. Maybe a disclaimer is in order?

**Comment by harding** [comment]
@luke-jr the page itself says, "Always remember that it is your responsibility to choose your wallet carefully and adopt good practices in order to protect your money" and the bottom of the page links to the current [legal disclaimer](https://bitcoin.org/en/legal), which covers that pretty well, I think.

**Comment by harding** [comment]
@MineForeman

> the not-so-good players with a pile of red marks beside their names.  Not opinions, just the facts.

Red marks *are* opinions.

> we should be informing users of common wallets (warts and all) instead of recommending something based on a set of arbitrarily decided upon best practice criteria (that's how i perceive it).

I think it's really hard for non-technical users to make intellegent comparative decisions about wallets. For example, FooWallet and BarWallet have the same marks except that FooWallet doesn't have HSTS and BarWallet doesn't have 2FA... so which one would my Grandmother choose?

Maybe we should explain what HSTS and 2FA are, so Grandma can make an informed choice. That might be reasonable for a binary decision, but how many non-technical users are going to read through a dozen or more pages of explanations for our various criteria?  It seems more likely to me that users will either look for a more useful page, or they'll simply choose one of the wallets with the least amount of red---even if one of those red marks is "broken CSPRNG".

By making (hopefully reasonable) technical decisions for the user, and by clearly explaining the decisions we didn't make in the security and privacy scores, I think we add a lot of value to the wallet selection process.

---

I really like the suggestion to create a page on the Wiki.  If you can create an informative resource there, even if it's basic, discussing it here might be much easier, and it would allow us to easily choose between either porting it here or simply linking to it from one or more of our pages.  (I should really add this paragraph to our README somewhere, as it's good general advice.)

**Comment by saivann** [comment]
I pretty much agree with @harding @luke-jr @gurnec and @gmaxwell in general.

For the record, I also think having an actual example that works outside of bitcoin.org would be a better starting point. For example, http://www.walletlist.org/ seemed like an interesting project, but at the first glance it seems like the project didn't scale. I worry about the maintainability of this idea.

I also wonder if the extra information displayed would be of any value if it's the same for all wallets, assuming we're already listing most popular wallets (with the exception of blockchain.info). In other words, at least for now, current requirements seem to be easily at reach for most wallets.

And to my experience, having wallet developers care about security issues that are otherwise easily de-prioritized for features is a nice side effect of having basic requirements and not just scores. There is perhaps 5-7 wallets now enabling HSTS because I took care to ask them. This isn't huge, but this didn't require a lot of effort either, and the result is a more secure ecosystem for everyone.

As for the disclaimers, to my experience and great despair, people don't read or make informed choice :) . Even though that's what I've been trying to encourage with the current layout and disclaimers.
